src/crab/ebpf_domain.cpp

```diff
@@ -643,8 +672,11 @@ void ebpf_domain_t::operator()(const Assume& s) {
         if (type_inv.same_type(m_inv, cond.left, std::get<Reg>(cond.right))) {
             m_inv = type_inv.join_over_types(m_inv, cond.left, [&](NumAbsDomain& inv, type_encoding_t type) {
                 if (type == T_NUM) {
-                    if (!is_unsigned_cmp(cond.op))
-                        for (const linear_constraint_t& cst : jmp_to_cst_reg(cond.op, dst.value, src.value))
+                    if (is_unsigned_cmp(cond.op))
+                        for (const linear_constraint_t& cst : unsigned_jmp_to_cst_reg(m_inv, cond.op, dst.value, src.value))
+                            inv += cst;
+                    else
+                        for (const linear_constraint_t& cst : signed_jmp_to_cst_reg(cond.op, dst.value, src.value))
```

Comment:
I'd use brackets for this kind of nesting.

Reply:
Done